### ISSUE TITLE
Upgrade to 0.10.0 fails

### ISSUE THREAD

**Issue by tfenster** [fermyon/cloud-plugin#218]
I got the following message when running a `spin cloud` command (it was `spin cloud variables set ...`, but I don't think that matters)

```
This plugin can be upgraded. Version 0.10.0 is available,
but may not be backward compatible with your current plugin.
To upgrade, run `spin plugins upgrade cloud`.
```

Hence I tried the upgrade command, but that leads to this error:

```
vscode ➜ /workspaces/verified-bluesky (tfe/only-verifiy-mode) $ spin plugins upgrade cloud
Error: Newer version 0.9.1 of plugin 'cloud' is already installed. To downgrade to version 0.10.0, run `spin plugins upgrade` with the `--downgrade` flag.
```

Here is a screenshot of the full flow

![image](https://github.com/user-attachments/assets/cc06b230-9d14-4107-a8db-74643e6c0762)

I guess the version comparison is broken?

**Comment by itowlson** [comment]
Yes, sorry, this is fixed in https://github.com/fermyon/spin/pull/2947 but that's not in a release yet.  For the time being you would need to uninstall-reinstall to upgrade.  I apologise for this!
